airflow/hooks/subprocess.py

```diff
@@ -0,0 +1,90 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+import signal
+from subprocess import PIPE, STDOUT, Popen
+from tempfile import TemporaryDirectory, gettempdir
+from typing import Dict, List, Optional
+
+from airflow import AirflowException
+from airflow.hooks.base import BaseHook
+
+
+class SubprocessHook(BaseHook):
+    """Hook for running processes with the ``subprocess`` module"""
+
+    def __init__(self) -> None:
+        self.sub_process = None
+        super().__init__()
+
+    def run_command(
+        self, command: List[str], env: Optional[Dict[str, str]] = None, output_encoding: str = 'utf-8'
+    ):
+        """
+        Execute the command in a temporary directory which will be cleaned afterwards
+
+        If ``env`` is not supplied, ``os.environ`` is passed
+
+        :param command: the command to run
+        :param env: Optional dict containing environment variables to be made available to the shell
+            environment in which ``command`` will be executed.  If omitted, ``os.environ`` will be used.
+        :param output_encoding: encoding to use for decoding stdout
+        :return: last line of stdout
```

Comment:
stderr is redirect to stdout I think so:

```suggestion
        :return: last line of output (on stderr or stdout)
```